guides/manual/1-overview.md

```diff
@@ -1,71 +1,151 @@
+
+```hbs
+  Hello {{@user.name}}!
+```
+
+By building around the same interface as the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API), WarpDrive makes powerful request management features like caching, deduping and data normalization feel simple to use. We pair this with [reactive control flow](./concepts/reactive-control-flow.md) to give apps the ability to declaratively derive states with safety.
+
+```glimmer-ts
+import { Request } from '@warp-drive/ember';
+import { findRecord } from '@ember-data/json-api/request';
+import { Spinner } from './spinner';
+
+export default <template>
+  <Request @query={{findRecord "user" "1"}}>
+    <:error as |error state|>
+      <h2>Whoops!</h2>
+      <h3>{{error.message}}</h3>
+      <button {{on "click" state.retry}}>Try Again?</button>
+    </:error>
+
+    <:content as |result|>
+      Hello {{result.data.name}}!
+    </:content>
+
+    <:loading as |state|>
+      <Spinner @progress={{state.completedRatio}} />
+      <button {{on "click" state.abort}}>Cancel?</button>
+    </:loading>
+  </Request>
+</template>
+```
+
+Web clients are like high-latency, remotely distributed, often-stale partial replicas of server state - but requests using WarpDrive utilize an [advanced relational cache](./5-caching.md) that simplifies these problems, solving them when it can and providing intelligent escape valves for when it can't so that no matter what you can quickly get the data you need in the right state.
+
+WarpDrive's reactive objects transform raw cached data into rich, reactive data. The resulting objects are immutable, always displaying the latest state in the cache while preventing accidental or unsafe mutation in your app. The output and [transformation](./concepts/transformation.md) is controlled by a simple JSON [ResourceSchema](./concepts/schemas.md).
+
+```ts
+import { withDefaults } from '@warp-drive/schema-record';
+
+store.schema.registerResource(
+  withDefaults({
+    type: 'user',
+    fields: [
+      { kind: 'field', name: 'firstName' },
+      { kind: 'field', name: 'lastName' },
+      { kind: 'field', name: 'birthday', type: 'luxon-date' },
+      {
+        kind: 'derived',
+        name: 'age',
+        type: 'duration',
+        options: { field: 'birthday', format: 'y' }
+      },
+      {
+        kind: 'derived',
+        name: 'fullName',
+        type: 'concat',
+        options: { fields: ['firstName', 'lastName'], separator: ' ' }
+      },
+    ]
+  })
+)
+```
+
+[Mutation](./concepts/mutations.md) is handled within controlled contexts. The data to edit is "checked out" for editing, giving access to a mutable version. Local edits are seamlessly preserved if the user navigates away and returns without saving, and the changes are buffered from appearing elsewhere in your app until they are also committed to the server.
+
+```ts
+import { Checkout } from '@warp-drive/schema-record';
+
+// ...
+
+const editable = await user[Checkout]();
```

Comment:
This might be confusing for junior devs. Can we provide an API like `await checkout(user)`?

Reply:
I dislike it for a different reason (requiring the import). I've considered adding `await store.checkoutRecord(user)` but checkout is an implementation detail of SchemaRecord currently, we'd have to consider whether this is a permanent addition of a feature that all potential record classes must support and if so what the contract is